### PR TITLE
feature/no-bucket

### PR DESCRIPTION
**Pull request by bwalsh**
This  PR:
  * initial support for adding files without pushing to bucket #47 
  * draft PR

### CODE REVIEW

**Comment by matthewpeterkort** [comment]
Might want to add some sort of warning on the g3t pull if the researcher is about to download many terrabytes of data by scp.

File: tests/integration/test_no_bucket.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+from click.testing import CliRunner
+from gen3_util.repo.cli import cli
+from gen3_util.common import PROJECT_DIRECTORIES
+
+CURRENT_DIR = pathlib.Path.cwd()
+
+
+def setup_module(module):
+    """Setup for module."""
+    print("setup_module      module:%s" % module.__name__)
```
Comment (matthewpeterkort) [inline tests/integration/test_no_bucket.py]: change print() statements to click.echo()

**Review by matthewpeterkort — COMMENTED** [review]
Should probably add a test that simulates the no bucket adds, and then runs a g3t pull to get the data via the symlinks.

Nevermind. There is a test for this.

File: gen3_util/repo/cli.py [hunk 2]
```diff
@@ -389,39 +389,45 @@ def project_rm(config: Config, project_id: str):
...
-            completed_path = push_.config.commit_dir() / "emptied.ndjson"
-            push_.published_timestamp = datetime.now()
-
-            with open(completed_path, "w") as fp:
-                fp.write(push_.model_dump_json())
-                fp.write("\n")
-            click.secho(
-                f"Updated {completed_path}",
-                file=sys.stderr, fg='green')
+            if in_project:
```
Comment (matthewpeterkort) [inline gen3_util/repo/cli.py:415]: What is the purpose of this if statement ? Why are all of the local state changes getting disabled every time?

**Comment by bwalsh** [comment]
> Might want to add some sort of warning on the g3t pull if the researcher is about to download many terrabytes of data by scp.

For scp, we simply echo the scp commands to the console.  We don't actually download them.
This may change in the future, but that is what we have for now.

https://github.com/ACED-IDP/gen3_util/blob/feature/no-bucket-2/gen3_util/repo/puller.py#L58-L67

File: gen3_util/repo/cli.py (hunk 2 above)
Comment (bwalsh) [reply gen3_util/repo/cli.py:415]: It enables a sysadmin to empty a project without having to clone it first. Will add comment.

Comment (matthewpeterkort) [reply gen3_util/repo/cli.py:415]: Ok see that now

**Review by lbeckman314 — APPROVED** [review]

**Pull request merged**